File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    for i in range(left, right + 1):
        cnt = 0
        for j in range(1, int(i**(1/2)) + 1):
            if i % j == 0:
                cnt += 1
                if j ** 2 != i:
                    cnt +=1
        if cnt % 2 == 0:
            answer += i
        else:
            answer -= i
    return answer

File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    function divisor_count(num) {
        let count = 0;
        for(let i=1; i <= parseInt(num**(1/2)+1); i++){
            if(num % i === 0){
                count+=1;
                if(i**2 !== num) count+=1;
            }
        };
        return count;
    };
    let answer = 0;
    for(let i=left; i <= right; i++){
        (divisor_count(i) % 2 === 0) ? answer+=i : answer-=i;
    };
    
    return answer;
}